modules/document_repository/jsx/uploadForm.js

```diff
@@ -0,0 +1,197 @@
+    }
+    return (
+      <div className="row">
+        <div className="col-md-8 col-lg-7">
+          <FormElement
+            name="docUpload"
+            fileUpload={true}
+            onSubmit={this.handleSubmit}
+          >
+            <h3>Upload a file</h3><br/>
+            <SelectElement
+              name="category"
+              label="Category"
+              options={this.state.Data.fieldOptions.fileCategories}
+              onUserInput={this.setFormData}
+              hasError={false}
+              required={true}
+              value={this.state.formData.category}
+            />
+            <SearchableDropdown
+              name="forSite"
+              label="Site"
+              placeHolder="Search for site"
+              options={this.state.Data.fieldOptions.sites}
+              strictSearch={true}
+              onUserInput={this.setFormData}
+              required={true}
+              value={this.state.formData.forSite}
+            />
+            <SelectElement
+              name="instrument"
+              label="Instrument"
+              options={this.state.Data.fieldOptions.instruments}
+              onUserInput={this.setFormData}
+              value={this.state.formData.instrument}
+            />
+            <TextboxElement
+              name="pscid"
+              label="PSCID"
+              onUserInput={this.setFormData}
+              value={this.state.formData.pscid}
+            />
+            <TextboxElement
+              name="visitLabel"
+              label="Visit Label"
+              onUserInput={this.setFormData}
+              value={this.state.formData.visitLabel}
+            />
+            <TextboxElement
+              name="version"
+              label="Version"
+              onUserInput={this.setFormData}
+              value={this.state.formData.version}
+            />
+            <TextareaElement
+              name="comments"
+              label="Comments"
+              onUserInput={this.setFormData}
+              value={this.state.formData.comments}
+            />
+            <FileElement
+              name="file"
+              id="docUploadEl"
+              onUserInput={this.setFormData}
+              label="File to upload"
+              required={true}
+              value={this.state.formData.file}
+            />
+            <ButtonElement label="Upload File"/>
+          </FormElement>
+        </div>
+      </div>
+    );
+  }
+
+/** *******************************************************************************
+ *                      ******     Helper methods     *******
+ *********************************************************************************/
+
+
+  /**
+   * Handle form submission
+   * @param {object} e - Form submission event
+   */
+  handleSubmit(e) {
+    e.preventDefault();
+      this.uploadFile();
+        this.setState({
+          formData: {}, // reset form data after successful file upload
+          uploadProgress: -1,
```

Comment:
Nothing is being done with `uploadProgress`, so I believe this line can be deleted.